docs/reference/configuration.md

```diff
@@ -0,0 +1,112 @@
+## Configuration Reference
+
+![Default](../../assets/overview.png)
+This is a diagram of how the theme is split between its components.
```

Comment:
I know I briefly asked earlier about the usefulness/accuracy of this diagram.

Do you still think it's a net positive to have this diagram even with its inaccuracies? 

I think what I'd prefer is maybe a `[!WARNING]` block at the top saying that this diagram is not up to date and potentially that contributions are welcome if someone wants to take the time to improve it? Suppose it could even be a separate issue if you're interested in that idea.